### PR TITLE
add support for inline svgs in js templates

### PR DESCRIPTION
**Pull request by sndrs**
javascript implementation of the scala helper in #7785.
- works as requirejs plugin
- svgs need to be passed to templates, which is bit clunky, but it means they should only appear in the source once
### usage

``` javascript
define([
    'inlineSvg!svgs/logo',
    'inlineSvg!svgs/share-twitter!icon',
    'inlineSvg!svgs/share-twitter!icon!my-class',
    'inlineSvg!svgs/logo!!my-class',
], function (
    logo,
    twitterIcon,
    twitterIconWithClass,
    logoWithClass
) {})
```

results in the following strings:

``` html
<span class="inline-logo"><svg>…</svg></span>
<span class="inline-share-twitter inline-icon "><svg>…</svg></span>
<span class="inline-share-twitter inline-icon inline-my-class"><svg>…</svg></span>
<span class="inline-logo inline-my-class"><svg>…</svg></span>
```

### CODE REVIEW

File: static/src/javascripts/components/requirejs-inline-svg/inlineSvg.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            var data = name.split('!'),
+
+                path = data[0],
+                imageType = data[1] || '',
+                cssClasses = (data[2] && data[2].split(",") || []),
+
+                dirs = path.split('/'),
+                fileName = dirs.pop();
+
+            text.get(req.toUrl(dirs.join('/') + '/' + imageType + '/' + fileName + '.svg'), function(svg) {
```
Comment (stephanfowler) [inline static/src/javascripts/components/requirejs-inline-svg/inlineSvg.js]: Probably harmless, but you get `//` in here if imageType is `''`

**Comment by stephanfowler** [comment]
:+1: nice

**Comment by phamann** [comment]
:+1: Like this.

But as we discussed, I want us to be very cautious of adding these SVG's to our `app.js` file, as we'd be defeating the point of all of this work, and essentially just moving the bloat from CSS to JS. Most of these icons won't be needed on every page. If we go down this route, can we ensure that the JS icons get loaded within [`core.js`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/core.js) and therefore will benefit from longer term caching.

**Comment by robertberry** [comment]
Nice work :+1:

**Pull request merged**